### DISCUSSION TITLE
Create breakout cables (SFP28) as child interfaces

### DISCUSSION THREAD

**Discussion by fziegert** [netbox-community/netbox#15360] [Ideas]
Hello everyone,
I have only worked with Racktables (Docion) so far and am now working with Netbox for the first time. I am enthusiastic about Netbox, but I need an important function for my work:
My problem is about breakout cable documentation.

I only use breakout cables in the DAC area. I have already found out that you can assign several connections to an interface and thus use breakout cables.

With our cabling, it is sometimes the case that not all interfaces are occupied by the breakout cable. Sometimes the second or third cable is free. Due to the current options in NetBox, I cannot clearly see that cable 2 or 3 is free.
It would be great if I could leave one cable (e.g. cable 2) free or give each part of the breakout cable its own name.

Another option might be to create child interfaces for each individual cable. Currently, however, it is only possible for a virtual interface to be a child interface. It would be great if I could select SFP28, for example.

For me it is important that I can see in the cable trace that it is actually one cable that splits into four. And secondly, that I can see exactly how many cables I have in use.

Can you help me if I have overlooked any features or would it be possible to implement something like this?
Thank you very much!

**Comment by Hvost7** [comment]
We have same issue

**Comment by jeremystretch** [comment]
Each interface on either end of a breakout cable is a physically discrete interface, operating independently of its peers. These must be modeled as physical interfaces in NetBox; a breakout cable can attach to multiple interfaces at either end.

**Reply by DanSheps** [reply]
Just to add to this, you can set the breakout interface as a "parent" interface and mark it as connected, but this also depends on the specificc implementation of each NOS how they handle the parent interface.

**Reply by fziegert** [reply]
How can I make sure that it is clear for the documentation that the third cable from the breakout is not connected and the fourth is?

This function is very important for our planning of a new cluster. 
Currently, I can only connect from one side of a breakout cable to several other interfaces. But I cannot clearly label which cable goes where from the breakout.

**Reply by DCAuto** [reply]
I'm having a similar issue with breakout cables in that I have the "parent" switch interface connected to 4 server interfaces. This works well but I can not get child interfaces working in a nice way.

I have tried creating 4 virtual interfaces with the physical as a parent. The issue is that a virtual interface can not be a member of a lag for some reason.

So now i have the 400G physical and (4) 100G physical interfaces that aren't linked together through a parent child relationship.

Has anybody worked out a solution for this?

**Comment by fziegert** [comment]
How can I make sure in my documentation that I know that the cable from the breakout cable with label number 2 is not wired, but the cable with label three is connected to the following interface?
Unfortunately, I have not yet found a way to clearly define this.

**Reply by candlerb** [reply]
There are no "cable types" in Netbox, so an unconnected cable endpoint isn't a thing.

You could create a dummy patch panel where you "connect" the unused terminations; but even then you wouldn't necessarily be able to identify *which* of the N endpoints each was, since there's no guarantee on the ordering of terminations.  Also, if you trace through a 1-to-N breakout cable you won't be able to tell which cable is which.

If you need that level of tracing, then right now I think you have to lie to Netbox. Next to your QSFP interface, create four fake SFP+ interfaces, and connect "cables" into each of these.  As you already observed, you can't link them via parent relationship to the QSFP, so you have to do it via some sort of naming convention. But then your breakout cable is actually shown as 4 separate cables, and they can't be bundled together except via label or description.

**Reply by fziegert** [reply]
This is exactly the way I use it in rack tables. 
But what might be an idea to rewrite it or add to it so that it is displayed as a breakout cable in the cable trace without losing its clear label.
Because entering it as you have described also distorts the cable quantity.
